lib/SILOptimizer/Analysis/ClosureScope.cpp

```diff
@@ -82,7 +82,8 @@ class ClosureScopeData {
   void recordScope(PartialApplyInst *PAI) {
     // Only track scopes of non-escaping closures.
     auto closureTy = PAI->getCallee()->getType().castTo<SILFunctionType>();
-    if (!isNonEscapingClosure(closureTy))
+    if (!isNonEscapingClosure(closureTy) ||
```

Comment:
It doesn't really make sense to me that both isNonEscaping and isCalleeDynamicallyReplaceable could be true.
For future reference, we should have a comment explaining how this came to pass so we can fix it with a proper assert later.